src/workerd/api/gpu/gpu-buffer.c++

```diff
@@ -46,7 +51,23 @@ GPUBuffer::getMappedRange(jsg::Optional<GPUSize64> offset,
 
   JSG_REQUIRE(ptr, TypeError, "could not obtain mapped range");
 
-  auto arrayBuffer = kj::arrayPtr((byte *)ptr, s);
+  struct Context {
+    jsg::Ref<GPUBuffer> buffer;
+  };
+  auto ref = JSG_THIS;
+  auto ctx = new Context{ref.addRef()};
```

Comment:
Since using `new` in unusual for our code base, please be sure to leave a comment here explaining why we're doing so. I know you have the comment there in the deleter callback but it is easy to miss.

Reply:
Added comments.